rclpy/src/rclpy/_rclpy.c

```diff
@@ -3629,6 +3629,57 @@ rclpy_get_rmw_qos_profile(PyObject * Py_UNUSED(self), PyObject * args)
   return pyqos_profile;
 }
 
+/// Manually assert that an entity is alive.
+/**
+  * When using RMW_QOS_POLICY_MANUAL_BY_*, the application must call this function at least as
+  * often as the qos policy liveliness_lease_duration.
+  * The passed entity can be a Publisher or a Node.
+  *
+  * Raises RuntimeError on failure to assert liveliness
+  * Raises TypeError if passed object is not a valid Publisher or Node
+  *
+  * \param[in] pyentity A capsule containing an rcl_node_t or rcl_publisher_t
+  * \return None
+  */
+static PyObject *
+rclpy_assert_liveliness(PyObject * Py_UNUSED(self), PyObject * args)
+{
+  PyObject * pyentity;
+
+  if (!PyArg_ParseTuple(args, "O", &pyentity)) {
+    return NULL;
+  }
+
+  if (PyCapsule_IsValid(pyentity, "rcl_node_t")) {
+    rcl_node_t * node = (rcl_node_t *)PyCapsule_GetPointer(pyentity, "rcl_node_t");
+    if (!node) {
```

Comment:
This guard is redundant and can be removed. `PyCapsule_IsValid` does the same checks as `PyCapsule_GetPointer`.